File: c/src/basic/files.c
#include "files.h"

static void crea_file_vuoto() {
  FILE *fptr;
  fptr = fopen("filename.txt", "w");
  fclose(fptr);
}

static void Scrivi_su_un_file_nuovo() {
  FILE *fptr;
  fptr = fopen("filename2.txt", "w");
  fprintf(fptr, "Some text");
  fclose(fptr);
}

static void leggi_file_e_stampa() {
  FILE *fptr;
  fptr = fopen("filename2.txt", "r");
  const int MAX=80;
  char stringa[MAX] ;

  while(fgets(stringa, MAX, fptr))
    printf("%s", stringa);

  fclose(fptr);
  printf("\n");
}

void files_handling() {
  crea_file_vuoto();
  Scrivi_su_un_file_nuovo();
  leggi_file_e_stampa();
}

File: rust/src/basic/files.rs
use std::sync::Mutex;
use std::io::{Write, Read};
//---------------------------------------------------------------------------------------
// Gestione concorrenza per input e file
//---------------------------------

//static INPUT_LOCK: Mutex<()> = Mutex::new(());
static FILES_LOCK: Mutex<()> = Mutex::new(());


//---------------------------------------------------------------------------------------
// Gestione file
//---------------------------------

use std::fs::File;
fn crea_file_vuoto() {
  let _lock = FILES_LOCK.lock().unwrap();
  let _ = File::create("filename.txt").expect("Unable to create file");
}

fn scrivi_su_un_file_nuovo() {
  let _lock = FILES_LOCK.lock().unwrap();
  let mut fptr = File::create("filename2.txt").expect("Unable to create file");
  write!(fptr, "Some text").expect("Unable to write data");
}

fn leggi_file_e_stampa() {
  let _lock = FILES_LOCK.lock().unwrap();
  let mut fptr = File::open("filename2.txt").expect("Unable to open file");
  let mut stringa = String::new();
  while fptr.read_to_string(&mut stringa).expect("Unable to read file") > 0 {
    print!("{}", stringa);
  }
  println!(); // Aggiunge una nuova riga
}

pub fn files_handling() {
  crea_file_vuoto();
  scrivi_su_un_file_nuovo();
  leggi_file_e_stampa();
}